src/amo/routes.js

```diff
@@ -3,9 +3,11 @@ import { IndexRoute, Route } from 'react-router';
 
 import App from './containers/App';
 import Home from './containers/Home';
+import DetailPage from './containers/DetailPage';
 
 export default (
-  <Route path="/(:lang/)" component={App}>
+  <Route path="/:lang/:application" component={App}>
     <IndexRoute component={Home} />
+    <Route path="addon/:slug(/)" component={DetailPage} />
```

Comment:
This can now have the optional trailing slash removed having move to a middleware.